### PR TITLE
Fixes issue with LaunchUIView not reacting to changes in auth status

### PR DESCRIPTION
**Pull request by vishnuravi**
- Modifies CKStudyUser to conform to ObservableObject and publishes updates to its currentUser property. 
- Adds a listener for auth status changes from Firebase which updates currentUser.
- Modifies LaunchUIView to use @AppStorage to react to completion of onboarding instead of NotificationCenter

The above changes resolve issue #126.

### CODE REVIEW

**Review by PSchmiedmayer — APPROVED** [review]
@vishnuravi Thanks a lot for the changes and improvements! 
The changes are a clear improvement! I had a few comments on how to improve the PR further. Let me know if you have any questions regarding the comments 👍

**Comment by PSchmiedmayer** [comment]
One further comment: There seems to be a small issue in the Xcode project configuration for the `CKUploadToGCPTaskViewControllerDelegate.swift` file: The path stored in the Xcode project is `/Users/vishnu/code/CardinalKit/CardinalKit-Example/CardinalKit/Library/ResearchKit/CKUploadToGCPTaskViewControllerDelegate.swift` which is a full path and not relevant to the group. I would suggest readding the file to make sure that the project also builds on other machines.

File: CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift [hunk 1]
```diff
@@ -9,26 +9,24 @@ import Foundation
 import Firebase
```
[the remark this answers is not in the document]
Comment (vishnuravi) [reply CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift:9]: Thanks for picking up on this! I think we should remove the one at /Library/Auth/CKStudyUser.swift and keep the one at Networking/CKStudyUser.swift.

File: CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift [hunk 2]
```diff
@@ -74,16 +72,34 @@ class CKStudyUser {
...
     }
+
+    init() {
+        // listen for changes in authentication state from Firebase and update currentUser
+        authStateHandle = Auth.auth().addStateDidChangeListener { [weak self] (auth, user) in
+            if user != nil {
+                self?.currentUser = user
+            } else {
+                self?.currentUser = nil
+            }
```
[the remark this answers is not in the document]
Comment (vishnuravi) [reply CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift]: This is a great suggestion, I've incorporated this change!

File: CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift [hunk 3]
```diff
@@ -9,26 +9,24 @@ import Foundation
...
     /* **************************************************************
      * the current user only resolves if we are logged in
-    **************************************************************/
-    var currentUser: User? {
-        // this is a reference to the
-        // Firebase + Google Identity User
-        return Auth.auth().currentUser
-    }
+     **************************************************************/
+    @Published var currentUser: User?
```
[the remark this answers is not in the document]
Comment (vishnuravi) [reply CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift]: This makes sense, thanks!

File: CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (vishnuravi) [reply CardinalKit-Example/CardinalKit/Library/Auth/CKStudyUser.swift:9]: Following up on this - as we had discussed on Slack, merging the two seems a bit more complex that initially thought. I will separate this into a different issue to work on.

**Comment by vishnuravi** [comment]
> One further comment: There seems to be a small issue in the Xcode project configuration for the `CKUploadToGCPTaskViewControllerDelegate.swift` file: The path stored in the Xcode project is `/Users/vishnu/code/CardinalKit/CardinalKit-Example/CardinalKit/Library/ResearchKit/CKUploadToGCPTaskViewControllerDelegate.swift` which is a full path and not relevant to the group. I would suggest readding the file to make sure that the project also builds on other machines.

As we discussed on Slack, it is odd that the build succeeded on Github Actions despite this. I will create a separate issue to address this.

**Pull request merged**